src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -735,7 +735,7 @@ static function (ChildDefinition $definition, AsPeriodicTask|AsCronTask $attribu
             $container->getDefinition('config_cache_factory')->setArguments([]);
         }
 
-        if (!$config['disallow_search_engine_index'] ?? false) {
+        if (!($config['disallow_search_engine_index'] ?? false)) {
```

Comment:
the ?? operator is not needed and confusing as you spotted
```suggestion
        if (!$config['disallow_search_engine_index']) {
```